app/src/src_nonfreenet/org/breezyweather/sources/nws/NwsService.kt

```diff
@@ -99,8 +101,10 @@ class NwsService @Inject constructor(
             .getOrElse(id) { null }?.getOrElse("gridX") { null }
         val gridY = location.parameters
             .getOrElse(id) { null }?.getOrElse("gridY") { null }
+        val station = location.parameters
+            .getOrElse(id) { null }?.getOrElse("station") { null }
 
-        if (gridId.isNullOrEmpty() || gridX.isNullOrEmpty() || gridY.isNullOrEmpty()) {
+        if (gridId.isNullOrEmpty() || gridX.isNullOrEmpty() || gridY.isNullOrEmpty() || station.isNullOrEmpty()) {
```

Comment:
It is only an issue if requesting `CURRENT`

Reply:
Will add extra condition.